tests/test_sample.py

```diff
@@ -151,3 +151,173 @@ def test_sample_exception_wrong_n(self, device, shots):
         # self.def.shots = 0, so this should also fail
         with pytest.raises(ValueError, match="The number of samples must be a positive integer"):
             dev.sample("PauliZ", [0], [], n=12.3)
+
+
+@pytest.mark.parametrize("shots", [8192])
+class TestTensorSample:
+    """Test tensor expectation values"""
+
+    def test_paulix_pauliy(self, device, shots, tol):
+        """Test that a tensor product involving PauliX and PauliY works correctly"""
+        theta = 0.432
+        phi = 0.123
+        varphi = -0.543
+
+        dev = device(3)
+        dev.apply("RX", wires=[0], par=[theta])
+        dev.apply("RX", wires=[1], par=[phi])
+        dev.apply("RX", wires=[2], par=[varphi])
+        dev.apply("CNOT", wires=[0, 1], par=[])
+        dev.apply("CNOT", wires=[1, 2], par=[])
+
+        dev._obs_queue = [
+            qml.PauliX(wires=[0], do_queue=False) @ qml.PauliY(wires=[2], do_queue=False)
+        ]
+
+        for idx in range(len(dev._obs_queue)):
+            dev._obs_queue[idx].return_type = qml.operation.Sample
+
+        res = dev.pre_measure()
+
+        s1 = dev.sample(["PauliX", "PauliY"], [[0], [2]], [[], [], []])
+
+        # s1 should only contain 1 and -1
+        assert np.allclose(s1 ** 2, 1, **tol)
+
+        mean = np.mean(s1)
+        expected = np.sin(theta) * np.sin(phi) * np.sin(varphi)
+        assert np.allclose(mean, expected, **tol)
+
+        var = np.var(s1)
+        expected = (
+            8 * np.sin(theta) ** 2 * np.cos(2 * varphi) * np.sin(phi) ** 2
+            - np.cos(2 * (theta - phi))
+            - np.cos(2 * (theta + phi))
+            + 2 * np.cos(2 * theta)
+            + 2 * np.cos(2 * phi)
+            + 14
+        ) / 16
+        assert np.allclose(var, expected, **tol)
```

Comment:
If you write a stochastic test, then the tolerance should depend on the number of shots, otherwise it is arbitrary.

Reply:
(Better not to write stochastic tests at all, I think settling to checking that the underlying function behaves as expected should be enough)

Reply:
Good point, I will modify `conftest.py` so that the 'stochastic' tolerance is shot dependent.

As a side note, I feel this is more of an integration test, as I wrote it so that it could be evaluated on hardware to ensure that the classical post-processing is doing what we would expect